### ISSUE TITLE
Dragging extension can be erratic

### ISSUE THREAD

**Issue by PonteIneptique** [alpheios-project/webextension#286]
- Video : [2020-05-07 10-03-03.mkv.zip](https://github.com/alpheios-project/webextension/files/4591553/2020-05-07.10-03-03.mkv.zip)
- Browser: Firefox
- Explanation (I think): dragging works well as long as the cursor stays in the bar. But if you move quickly, as soon as your cursor (still clicked) move out of the "div", the window does not move anymore but your cursor stays in dragging position.

**Comment by balmas** [comment]
@kirlat not sure if we can do anything about this if the source of the error is interact.js but can you look into it a little?

**Comment by kirlat** [comment]
I definitely will. Interact.js went through multiple updated and is now at version 1.9.13 (production build uses 1.6.3, I believe) so I am hoping maybe they've fixed it.

**Comment by kirlat** [comment]
@PonteIneptique, thanks for the video, it's very helpful! I can reproduce the issue in Firefox on Windows, will see if we can do anything about it.

**Comment by PonteIneptique** [comment]
Anything for @alpheios-project/admins :dancers:

**Comment by kirlat** [comment]
@balmas: I did an investigation and I think the problem lays in `interact.js`; I have no better ideas at the moment. There is a callback in our code that `intaract.js` calls whenever a mouse pointer is moved when in a drag mode. This callback gets horizontal and vertical move distances that `interact.js` calculates. We use this data to adjust a position of a popup.

It seems that this data is delayed in FF. I've tried different settings but none has any effect. What's interesting, it works perfectly in Chrome. This, it seems, proves that it's something within `interact.js` because it's the same code running in both FF and Chrome yet the results are different.

I've inspected `interact.js` docs and issues but found nothing that could explain it. Information is very scarce there. I'm not sure we'll be able to solve it without help of `interact.js` developers, and that might take a long time.

I think our best option here is probably switch from `interact.js` to native HTML Drag and Drop API. It should not be too hard as its callbacks are pretty similar to what we use. We can switch to it in FF only, or we can use it in all browsers. But if we decide to do it, it should probably be done after the release.

What do you think?

**Comment by balmas** [comment]
agree, both on trying to eliminate the interact.js dependency and on waiting until after the release. I'll move this issue to alpheios-core and take it out of the release.
